src/base/settingvalue.h

```diff
@@ -41,17 +40,17 @@
 template <typename T>
 class CachedSettingValue
 {
-    using ProxyFunc = std::function<T (const T&)>;
-
 public:
     explicit CachedSettingValue(const char *keyName, const T &defaultValue = T())
         : m_keyName(QLatin1String(keyName))
         , m_value(loadValue(defaultValue))
     {
     }
 
+    // ProxyFunc = T (*)(const T&)
+    template <typename ProxyFunc>
```

Comment:
Another attempt to make it that way... Is the gain so great that you are willing to sacrifice the best compilation error handling?

Reply:
IMO the pretty error message (if that is what you meant) is rarely seen, because the interface is quite simple often this type checking can easily be done by hand and seldom changed afterwards.
Even C++ spec has written such "requirements" which cannot be checked by compiler easily before C++20: https://en.cppreference.com/w/cpp/named_req

I don't meant to push this change hard, just something I think we can afford without sacrificing others too much.

Reply:
>if that is what you meant

I mean the following: if you currently pass incorrect proxy func (with incorrect signature) the compiler says "your code is wrong", after your changes it says "CachedSettingValue code is wrong".

Reply:
>I mean the following: 

Quoting from the cpp reference link:
>Until then, the burden is on the programmer to ensure that library templates are instantiated with template arguments that satisfy these requirements. Failure to do so may result in very complex compiler diagnostics.

It is a known problem in C++ and I think it is something tolerable.

Reply:
Okay, whatever... I just wanted to make life easier for the other contributors.

Reply:
Please change the comment. ProxyFunc isn't supposed to be the pointer-to-function only.
E.g.:
>The signature of the ProxyFunc should be equivalent to the following: T proxyFunc(const T &a);

Reply:
Done.